src/ir/mod.rs

```diff
@@ -894,6 +903,24 @@ impl<'expr> Visit<'expr> for ExprId {
     }
 }
 
+impl VisitMut for ExprId {
+    fn visit_mut<V>(&mut self, visitor: &mut V)
+    where
+        V: VisitorMut,
+    {
+        // TODO: this is somewhat unfortunate and seems like it's susceptible to
+        // being bug-prone when we have a DAG of an IR. Is there perhaps a
+        // different pattern we should be using? Maybe remove methods from
+        // `VisitMut` or something like that?
```

Comment:
At that point we will want to just have a graph traversal where nodes are `ExprId` and let folks do the deref + match manually I think.

Reply:
Ah good point, I'll add this to the comment here